### DISCUSSION TITLE
split dataset according to percentage

### DISCUSSION THREAD

**Discussion by J-You** [deepmodeling/deepmd-kit#1742] [Q&A]
Hi all,
I try to use the following code to split dataset:
```
from dpdata import LabeledSystem
outcar_file = ['OUTCAR_0700', 'OUTCAR_0600']

total_system=LabeledSystem()
for f in outcar_file:
    ls=LabeledSystem(f, fmt='vasp/outcar', begin=0, step=1)
    total_system.append(ls)
total_system.shuffle()

split_num = len(total_system) * 0.8
total_system[:split_num].to_deepmd_npy('../data/training_set', set_size=10000)
total_system[split_num:].to_deepmd_npy('../data/validation_set', set_size=10000)
```
The suffix of OUTCAR means temperature. I use 80% of `total_system` as training set, the rest as validation set. I am not sure if this code is correct. Could you please check it or give me some advancing suggestion?

Thanks,
Jiang

**Comment by njzjz** [comment]
Your `split_num` is `float`. It should be converted to `int`.

**Reply by J-You** [reply]
Thanks for your suggestion.
